### PR TITLE
Updated test shell project

### PR DESCRIPTION
**Pull request by zkasmi**

### CODE REVIEW

File: test_shell/ReadMe.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+This project shows how to use own or the system shell commands. In order to use the system shell commands:
+1. Additionally to the module: shell, shell_commands, uart0 and posix, the module for the corresponding system command is to include, e.g. module ps 
+for the ps command (cf. the Makefile in the project root directory). 
+2. the shell must be initialized as follows:
+	2.1 shell_t sys_shell;
+	2.2 shell_init(&sys_shell, NULL, shell_readc, shell_putchar);
```
Comment (OlegHahm) [inline test_shell/ReadMe.txt]: I would suggest:

```
2.2 shell_init(&shell, shell_commands, shell_readc, shell_putchar); 
      or
      shell_init(&sys_shell, NULL, shell_readc, shell_putchar); /* to initialize without the built-in shell commands */
```

File: test_shell/main.c [hunk 2]
```diff
@@ -1,53 +1,81 @@
...
-    {NULL, NULL, NULL}
-};
+const shell_command_t shell_commands[] = { { "start_test", "starts a test",
+		print_teststart }, { "end_test", "ends a test", print_testend }, {
+		"start_own_test", "starts own test", print_own_test }, { NULL, NULL,
+		NULL } };
 
 int main(void) {
-    printf("test_shell.\n");
+	//printf("Moin. build on %s %s SVN-Revision: %s\n", kernel_builddate, kernel_buildtime, kernel_svnrevision);
```
Comment (OlegHahm) [inline test_shell/main.c]: Delete this line.

File: test_shell/main.c [hunk 3]
```diff
@@ -1,53 +1,81 @@
...
 
-const shell_command_t shell_commands[] = {
-    {"start_test", "", print_teststart},
-    {"end_test", "", print_testend},
-    {NULL, NULL, NULL}
-};
+const shell_command_t shell_commands[] = { { "start_test", "starts a test",
+		print_teststart }, { "end_test", "ends a test", print_testend }, {
+		"start_own_test", "starts own test", print_own_test }, { NULL, NULL,
+		NULL } };
```
Comment (OlegHahm) [inline test_shell/main.c]: Context help is useful, but I prefer the former formatting.

**Comment by OlegHahm** [comment]
Nice work, but the indentations do not match the coding conventions. (Though the code did not comply the coding conventions anyway, I would avoid to make the formatting worse.)

File: test_shell/main.c [hunk 4]
```diff
@@ -1,53 +1,90 @@
...
 }
 
-void print_testend(char* str) {
+void print_testend(char *str)
+{
     printf("[TEST_END]\n");
 }
 
-int shell_readc() {
+void print_own_test(char *str)
```
Comment (OlegHahm) [inline test_shell/main.c:44]: What's this for?

File: test_shell/main.c [hunk 5]
```diff
@@ -1,53 +1,90 @@
...
     shell_t shell;
     shell_init(&shell, shell_commands, shell_readc, shell_putchar);
-
     shell_run(&shell);
 
+    //init system shell
+    //shell_t sys_shell;
+    //shell_init(&sys_shell, NULL, shell_readc, shell_putchar);
+    //shell_run(&sys_shell);
+
```
Comment (mehlis) [inline test_shell/main.c:87]: in or out?

Comment (OlegHahm) [inline test_shell/main.c:87]: I think the idea here was to show that the shell can be also initialized without an own shell_commands struct.

**Comment by OlegHahm** [comment]
Manually merged.

**Pull request closed without merging**